### PR TITLE
[GSoC] Added 240-yolox notebook

### PR DESCRIPTION
**Pull request by sawradip**
### This  notebook is done as prerequisite of GSoc'23. 

It is usage of YOLOx model with OpenVino, and quick inference  with CPU.

I  was assigned this task in #832.

Though there is a Notebook naming gap currently **204**,  I  have skipped that as  I have another  notebook submitted that is PR #814, waiting for review.

### CODE REVIEW

**Comment by sawradip** [comment]
@adrianboguszewski The test raised error at some point, that I didn't even touch, and at my own (fork to fork) testing, this test was passed. Can you shed some light on  this?

**Comment by adrianboguszewski** [comment]
@sawradip, don't worry about it if it's not related to your contribution

**Comment by eaidova** [comment]
@sawradip please start from initial framework and using original repository not from onnx. The goal of such tutorials not just show demo, but also help people who want run inference on own yolox based models using openvino. Usually after training people do not have onnx itself

**Comment by eaidova** [comment]
and also your tutorial uses legacy openvino API 1.0, it is also not good showcase, because it will be removed very soon. Please migrate  on API 2.0.
You can find more information how to do it in openvino documentation:
https://docs.openvino.ai/latest/openvino_2_0_transition_guide.html

**Comment by sawradip** [comment]
Thanks a lot for the feedback. I am sorry that I wasn't  aware  that I should have started from the beginning. Also thanks for pointing  out  the versioning issue. I will fix all of those.

**Comment by sawradip** [comment]
@eaidova  I had a confusion.  I can re-organize the notebook in two ways.
1. Bringing the complete implementation inside the notebook, no dependency or import form the YOLOX repository. In that case the notebook will contain a lot of code(almost the full implementation)
2. Importing the necessary classes and functions into the notebook, adding the YOLOX repository in the system path. In than case, notebook will be comparatively lightweight, as well as more accessible to the users. 

I can do option 1 or option 2 or both. Can you hint me to the expected approach?

**Comment by eaidova** [comment]
@sawradip I prefer cloning repo inside notebook and work with it, if it will requires to add in to path, then ok. Please look on example notebook provided in issue, it makes similar things. No need reimplement functions and classes from repo

**Comment by sawradip** [comment]
@eaidova  I have submitted my improvements. If you could give your feedback, whenever you get time.

**Comment by sawradip** [comment]
Can anyone of you kindly approve the checks to run, when you are free?

@adrianboguszewski  @eaidova

**Comment by eaidova** [comment]
@sawradip ci checks do not run not due to lack of approval, you have merge conflicts which should be resolved before

**Comment by sawradip** [comment]
@eaidova  I am extremely sorry that I wasn't aware about that fact. I didn't know the tests are held due to merge conflicts. That is why I didn't update mush, as I thought maintainers were busy due to heavy workload, and will approve testing when they are available, and then provide me feedback.

**Comment by eaidova** [comment]
> @eaidova I am extremely sorry that I wasn't aware about that fact. I didn't know the tests are held due to merge conflicts. That is why I didn't update mush, as I thought maintainers were busy due to heavy workload, and will approve testing when they are available, and then provide me feedback.

@sawradip do not worry, everything is fine :) yes, as you noted maintainers are busy right now, hopefully, there is a lot of time for the submission deadline, I'll try to find some time slot for review on weekends

**Comment by adrianboguszewski** [comment]
@sawradip, we will review it when we find the time. Don't worry, it has to be open by April 4, but it doesn't have to be reviewed or merged :)

**Comment by sawradip** [comment]
Thanks a lot for your reviews. I  will be back with the suggested  improvements  shortly.

**Comment by sawradip** [comment]
@eaidova Hello! I have added all your suggested improvements.

To summarize, 
* Organized and used minimal environment. Turns out that, with some workarounds, we can avoid most of the external installation. And use only the ones available from the repository `requirements.txt` or already installed in google colab. 
* Added original project paper and repository link in Notebook description as well as corresponding Readme file,
* Removed some unnecessary imports.
* Removed the plots at the end.
* Removed mentioning 2.0 at the header.
* Loaded the large numpy array to an `.npy` file.

**Comment by adrianboguszewski** [comment]
Please also do the self-check:
- [ ] each function is described by docstrings and type hints
- [ ] notebook contains explicit descriptions and explanatory diagrams
- [ ] there is a README.md file in consistent style (look at other notebooks)
- [ ] the notebook is added to the main README and notebooks/README files
- [ ] there are no grammar, interpunctions or typo issues (use any free tool for that e.g. Grammarly)
- [ ] there are no committed files besides notebook and readme (please use images or videos from data dir)
- [ ] all CI checks passed

File: notebooks/227-whisper-subtitles-generation/227-whisper-subtitles-generation.ipynb [hunk 1]
```diff
@@ -120,10 +120,10 @@
     "mel = torch.zeros((1, 80, 3000))\n",
```
Comment (eaidova) [inline notebooks/227-whisper-subtitles-generation/227-whisper-subtitles-generation.ipynb:120]: <pre class="ql-syntax" spellcheck="false">Line #39.        Extract audio signal from a given video file, then convert it to float,
</pre><p>please remove unrelated changes from your submission</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='24'/>

File: notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb [hunk 2]
```diff
@@ -0,0 +1,4585 @@
+{
```
Comment (eaidova) [inline notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb:1]: <p>in some cases it can be useful to  provide model performance on small subset without accuracy testing on whole dataset, could you please add step for benchmarking model performance as separated step from accuracy evaluation?</p><p>Users may want to see performance benefit from quantization without long lasting accuracy checking. you can use benchmark_app for these purpoces</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/845/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='60'/>

File: notebooks/227-whisper-subtitles-generation/227-whisper-subtitles-generation.ipynb (hunk 1 above)
Comment (sawradip) [reply notebooks/227-whisper-subtitles-generation/227-whisper-subtitles-generation.ipynb:120]: Done.Reverted that commit.

File: notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb (hunk 2 above)
Comment (sawradip) [reply notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb:1]: `benchmark_App` testing added. Looking forward to any other suggestion you might have. @eaidova

Comment (sawradip) [reply notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb:1]: @eaidova There is some bug in `227-whisper-subtitles-generation` notebook that is failing ci tests, as you might know already. For example you can see my *nbval / build_nbval (ubuntu-20.04, 3.8) (pull_request)* test is stuck, that has cancelled other tests.

Without all CI tests running is there any other way to know if there was any CI test triggering error in my own notebook, as I am going through incremental changes in my notebook? 

And as @adrianboguszewski  has mentioned,  as a participants all my CI checks should be passed. 
- [ ] all CI checks passed

I hope a solution or clarification might be helpful for the GSOC'23 participants. Thank you.

**Comment by sawradip** [comment]
I am sorry that, I don't know how to resolve this conflict. The addition of my notebook name causes this error, and if I remove that, it raises CI error.  @eaidova

File: notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb [hunk 3]
```diff
@@ -0,0 +1,4612 @@
...
...pTLtpMFZz//hI8EHg+8IaPB5PjMOAMhpbOZnGQiSnQEhpDTuk7KRk8ZFRJO6Oj3z25ef83u/+jghvE7DEX2e1fg85B2Y/EkMqyEosjFFJDTXaYBpHKuagIUQUkmElU0+k65qCsOmy7BcBrrOWqpKpsHEVzrVUTYUx4kwRQubh4fBrz/FvdZFJKfHmzZu1w1lwQGFDLB1ntd74S+dl7cVCZdmpLNOCaA8cWklE6tJ1LQ/QMrksFvbGWGIKWCW2F7OfVvhmnkemaSjc84xzBue68qBoUhSvrugTkSCJhI2hbRuZxowsLlvb4FMSTn1dk5Moyi9eZE9giyz8d6M1KhvmSfYHk/ec+7ccDyfmcWb2k0w1lWO33XB9teHFzR5rHVOh7U4R5tmSrj/gqDf8/F3PT/74Nb/6/GccHo9E30NKaD+wyQc+bOHl1nBrFTo88PjqtRRU67BVUyAJOVyUMXSbhraRQ10IDIJpbxsRvCUlLK26btBGMU2jFJVFbFrVjFMWgWGKhFnsO3yMkveDYhgHuq4j5cQwCGEiKYhBrNe1syu9dyF01HXN4XBcJ9rzOKz07Wmayt6qoes2aKvp6oZaW+7uXuO90ITrpsbUhpgUOUbu3r3m0y9ec/d4oG1q2lbMNVcWlRIW2ul44vrZc5wVsopVGucM1iqaWijylatpCx3blAN0CXUjJRE1GkWMYsioUesS3UexOfLztFoZGWPXyX6BcL2X+3sqE4k2TyN8L7TqynwznmGxDbrsNBZjy8WDSyAx+TtqdTNYCAkO1gZx0aItfz7nxBJbsbDgQGyMBLoUavfyvC/kgpyl2dRaFThdDE5l9y3Pqi0sRWlGZVJaDmRpRl1R1ANcJhmtJYrDWs04DMIGQ1FXDZUVQsocZs76zDA6Tj5gq4qcMuP5zBwCrq4wWpGiJ6fi/K5AsQhmk0SRZdYCq6SO8dknn/P9737E82c3l2kwBoYSqyE07yIoTdJEbbdbQpIdUioT78JUy1H2bHEhvyTWTCCUiChMXdPVLfM4MIwnDseZv/R7v4NWmnGaOJ0GxnHi4XDP+TzyxRdvfu05/q0uMgJjdati/XQ6FWphtdqnLEv55aF4qjjWWpcHnUL9RLQqYbHVVvK5jLglwwUDXv6bCiarlKVta26aK6H7nQcy+RtU5wXzr6pKnJcRHnsIkbppZWFtNBjFbr+lkEp
```
Comment (zhuo-yoyowz) [inline notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb:2032]: It is not recommended to include the model size comparison here, since the model size itself is not large in its raw format. Could focus on performance comparison only, with benchmark test results.

Comment (zhuo-yoyowz) [inline notebooks/240-yolox-object-detection/240-yolox-object-detection.ipynb:2480]: "More" is not needed here, as we compare performance with benchmark test only, which makes more sense than comparing model size in this case.
